src/main/java/picard/util/GraphUtils.java

```diff
@@ -49,7 +49,8 @@
             IntStream.range(0, neighbors.size()).forEach(i ->
                     neighbors.get(i).stream().forEach(j -> joinNodes(cluster, j, i)));
 
-            return nodes.stream().collect(Collectors.toMap(n -> n, n -> cluster[nodes.indexOf(n)]));
+            // Must call findRepNode() here in case nodes are orphaned and become only transitively equal to roots in their cluster.
+            return nodes.stream().collect(Collectors.toMap(n -> n, n -> findRepNode(cluster, cluster[nodes.indexOf(n)])));
```

Comment:
given that you're calling findRepNode, can you drop the cluster[.] call? 
```suggestion
            return nodes.stream().collect(Collectors.toMap(n -> n, n -> findRepNode(cluster, nodes.indexOf(n))));
```